### ISSUE TITLE
Dockerfile Alpine not working

### ISSUE THREAD

**Issue by akanshgulati** [aerospike/aerospike-client-nodejs#332]
Please update Dockerfile.alpine for latest Alpine(3.10.x) and Aerospike configuration.

**Comment by jhecking** [comment]
@akanshgulati Can you try out the updated Dockerfile.alpine from #333 and let me know if that works for you.

**Comment by akanshgulati** [comment]
Tried using it exactly same, Getting this error on build

<img width="975" alt="Screenshot 2020-01-02 at 15 17 12" src="https://user-images.githubusercontent.com/5287100/71661074-154bce80-2d73-11ea-9cd5-cdd0c25c482e.png">


It works if I manually delete the mentioned file but I don't think that's the right way.  Please verify.

**Comment by jhecking** [comment]
Weird. I am able to reproduce this. But it seems to happen only once every couple of builds, seemingly at random. Not really sure what's going on. Will have to check with the developer of the C client, since that's where the issue occurs.

Can you please try again, just to confirm whether you run into this issue every time or just occasionally, like me?

**Comment by akanshgulati** [comment]
While building, run `docker build . --no-cache` to reproduce this problem. I think that will reproduce it everytime.

**Comment by jhecking** [comment]
> @jhecking While building, run `docker build . --no-cache` to reproduce this problem. I think that will reproduce it everytime.

Yes, you would think so. I did, too. :-) But that's not what's happening. I did run that exact command twice, and the build only failed on the first run, but went through fine on the second run. 

Logs attached:

[build-logs.zip](https://github.com/aerospike/aerospike-client-nodejs/files/4015569/build-logs.zip)

**Comment by akanshgulati** [comment]
Try to add tagging, its coming everytime, again let's try to find the right solution here. Not very familiar with Docker. 

`docker build -t aerospike:v2 -t aerospike:latest ./ --no-cache`

**Comment by jhecking** [comment]
The tagging happens after the build, so won't affect this issue. It is indeed a race condition and the best work-around for now is to remove the offending `luaconf.h` file before the build. I've made the necessary changes in https://github.com/aerospike/aerospike-client-nodejs/pull/333/commits/4df4db15d51047618e375a730dbebd5ede27456c.

The next release of the C client SDK will have a proper fix in its Makefile.

Please give the updated Dockerfile another try.

**Comment by akanshgulati** [comment]
It worked but another issue came up while starting the node server. 
This is the further configuration we have: 

<img width="568" alt="Screenshot 2020-01-03 at 12 13 39" src="https://user-images.githubusercontent.com/5287100/71710506-9ada8780-2e22-11ea-847a-0a53251c057c.png">

**Error**

<img width="1102" alt="Screenshot 2020-01-03 at 12 15 09" src="https://user-images.githubusercontent.com/5287100/71710525-b6de2900-2e22-11ea-8f04-e9ef389e53c5.png">

**Comment by jhecking** [comment]
Do you include the `aerospike` package in your `package.json` file as well? And do you use a `yarn.lock` file? I'm just wondering whether the `RUN yarn install` step might overwrite some of the files in the existing `node_modules/aerospike` directory.

**Comment by jhecking** [comment]
Have you been able to resolve this problem?

FYI: I've updated the Dockerfile in #333 once more to update the Node.js / C client versions to the latest releases 3.14.0 / 4.6.10. This removes the need for the work-around for the luaconf.h build problem.

**Comment by akanshgulati** [comment]
Thanks @jhecking . Could not check it, will check this week and close this issue if all goes well. I am just wondering now because I have used **Ubuntu OS** and all works so nicely on CI and development side. 

Just want to compare if Alpine is even useful because it felt quite same because we are installing so many packages ourselves just to compile it and maintain it for new versions. Let me know if you have any findings in this regard.

**Comment by jhecking** [comment]
The extra steps are necessary because we don't currently provide pre-build, binary releases of the Aerospike C client for Alpine Linux. If Ubuntu works well for you, then that will definitely be easier to maintain over time.

**Comment by jhecking** [comment]
I've merged the updates to Dockerfile.alpine in #333 and am closing this issue for now.
